### PR TITLE
fix(e2core): Deal with dead processes

### PR DESCRIPTION
**Pull request by javorszky**
Related to #426 

This is a quality of life improvement that deals with crashed processes and recovery.

When using the `command.Start()` function call, the process starts in the background. We're supposed to use the `command.Wait()` to determine when either the process finished, or exited for some other reason.

This changeset accommodates collecting the wait functions, starts them for each sub process, and when they unblock, their ports are added to a `died` map, which will then be  removed automatically from watcher's bookkeeping.

I've tested this manually by:
1. creating an image for `e2core` using `make docker/dev`
2. starting up everything with se2's `docker compose up` (this is internal)
3. start following the logs with `docker compose logs e2core --follow` (this is internal)
4. making sure that I can execute an existing function
5. hopping into the e2core docker container with `docker exec -i -t a063f35d5c2b /bin/bash` where the `a06...` is the image ID you get if you list the images with `docker ps` for e2core
6. once inside, listing the process IDs with `ls -l /proc/*/exe`, which will have a similar output:
   ```
   lrwxrwxrwx 1 e2core e2core 0 May  3 17:20 /proc/1/exe -> /usr/local/bin/e2core
   lrwxrwxrwx 1 e2core e2core 0 May  3 17:21 /proc/16/exe -> /usr/local/bin/e2core
   lrwxrwxrwx 1 e2core e2core 0 May  3 17:31 /proc/38/exe -> /bin/bash
   lrwxrwxrwx 1 e2core e2core 0 May  3 17:33 /proc/self/exe -> /bin/ls
   lrwxrwxrwx 1 e2core e2core 0 May  3 17:33 /proc/thread-self/exe -> /bin/ls
   ```
7. of those, the second e2core is the sub process with process id 16
8. `kill -9 16` to terminate that
9. look at the logs to see that the termination and reaping of instance was completed and a new one was started during reconcile step
10. check again that I can execute the same function by sending the post request to the local edge endpoint

### CODE REVIEW

File: e2core/backend/satbackend/orchestrator.go
Comment (ospencer) [inline e2core/backend/satbackend/orchestrator.go:169]:
```suggestion
						ll.Err(err).Str("moduleFQMN", module.FQMN).Str("port", port).Msg("adding the port to the died list")
```

**Review by ospencer — APPROVED** [review]
LGTM. I would probably call it a `dead` list instead of a `died` list, but that's not enough for me to block this change.

File: e2core/backend/satbackend/watcher.go
Comment (cohix) [inline e2core/backend/satbackend/watcher.go:123]: It's hard to tell from the diff... why are we waiting on this?

**Comment by javorszky** [comment]
> I would probably call it a dead list instead of a died list

heck yeah smple past vs past participle! lemme change that around, it's smol effort

File: e2core/backend/satbackend/watcher.go
Comment (javorszky) [reply e2core/backend/satbackend/watcher.go:123]: when a new process is started with `exec.Start()`, one of the things that return from that function is a wait function.

That wait function is then placed in a goroutine which blocks on the specific command's (process's) wait, and only decrements the waitgroup once it gets past it.

This wait waits until all of the instances are terminated, which means all of the wait functions unblock, the wait group reaches zero.

At the end of the day it cleans up all the goroutines, so we won't have orphan goroutines that aren't attached to anything any more because the world was pulled out from under them. Or in other words, their parent process existed / finished.

File: e2core/backend/satbackend/watcher.go
Comment (callahad) [inline e2core/backend/satbackend/watcher.go:85]: `"exising"`?

File: e2core/backend/satbackend/watcher.go
Comment (cohix) [reply e2core/backend/satbackend/watcher.go:123]: Heh I meant 'in what scenario would we want to wait on this' but I see now it's when we want to terminate the entire watcher. That makes sense.

**Review by cohix — APPROVED** [review]

File: e2core/backend/satbackend/watcher.go
Comment (javorszky) [reply e2core/backend/satbackend/watcher.go:85]: mr T would be sad... fixing. Thank you!

**Pull request merged**
